context-propagation-quickstart fails with  Unsatisfied dependency for type org.jboss.resteasy.spi.HttpRequest

### Describe the bug

context-propagation-quickstart fails with  Unsatisfied dependency for type org.jboss.resteasy.spi.HttpRequest

```
[ERROR] Errors:
[ERROR]   PriceTest.test » Runtime java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.arc.deployment.ArcProcessor#validate threw an exception: jakarta.enterprise.inject.spi.DeploymentException: Found 2 deployment problems:
[1] Unsatisfied dependency for type org.jboss.resteasy.spi.HttpRequest and qualifiers [@Default]
	- java member: org.jboss.resteasy.plugins.providers.JaxrsServerFormUrlEncodedProvider#request
	- declared on CLASS bean [types=[jakarta.ws.rs.ext.MessageBodyReader<jakarta.ws.rs.core.Form>, org.jboss.resteasy.plugins.providers.JaxrsServerFormUrlEncodedProvider, java.lang.Object], qualifiers=[@Default, @Any], target=org.jboss.resteasy.plugins.providers.JaxrsServerFormUrlEncodedProvider]
[2] Unsatisfied dependency for type boolean and qualifiers [@Default]
	- java member: org.jboss.resteasy.plugins.providers.JaxrsServerFormUrlEncodedProvider():useContainerParams
	- declared on CLASS bean [types=[jakarta.ws.rs.ext.MessageBodyReader<jakarta.ws.rs.core.Form>, org.jboss.resteasy.plugins.providers.JaxrsServerFormUrlEncodedProvider, java.lang.Object], qualifiers=[@Default, @Any], target=org.jboss.resteasy.plugins.providers.JaxrsServerFormUrlEncodedProvider]
	The following beans match by type, but none have matching qualifiers:
		- Bean [class=java.lang.Boolean, qualifiers=[@ConfigProperty, @Any]]
	at io.quarkus.arc.processor.BeanDeployment.processErrors(BeanDeployment.java:1403)
	at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:305)
	at io.quarkus.arc.processor.BeanProcessor.initialize(BeanProcessor.java:156)
	at io.quarkus.arc.deployment.ArcProcessor.validate(ArcProcessor.java:526)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
	at java.base/java.lang.Thread.run(Thread.java:833)
	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
	Suppressed: jakarta.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type org.jboss.resteasy.spi.HttpRequest and qualifiers [@Default]
	- java member: org.jboss.resteasy.plugins.providers.JaxrsServerFormUrlEncodedProvider#request
	- declared on CLASS bean [types=[jakarta.ws.rs.ext.MessageBodyReader<jakarta.ws.rs.core.Form>, org.jboss.resteasy.plugins.providers.JaxrsServerFormUrlEncodedProvider, java.lang.Object], qualifiers=[@Default, @Any], target=org.jboss.resteasy.plugins.providers.JaxrsServerFormUrlEncodedProvider]
		at io.quarkus.arc.processor.Beans.resolveInjectionPoint(Beans.java:483)
		at io.quarkus.arc.processor.BeanInfo.init(BeanInfo.java:581)
		at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:293)
		... 13 more
	Suppressed: jakarta.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type boolean and qualifiers [@Default]
	- java member: org.jboss.resteasy.plugins.providers.JaxrsServerFormUrlEncodedProvider():useContainerParams
	- declared on CLASS bean [types=[jakarta.ws.rs.ext.MessageBodyReader<jakarta.ws.rs.core.Form>, org.jboss.resteasy.plugins.providers.JaxrsServerFormUrlEncodedProvider, java.lang.Object], qualifiers=[@Default, @Any], target=org.jboss.resteasy.plugins.providers.JaxrsServerFormUrlEncodedProvider]
	The following beans match by type, but none have matching qualifiers:
		- Bean [class=java.lang.Boolean, qualifiers=[@ConfigProperty, @Any]]
		at io.quarkus.arc.processor.Beans.resolveInjectionPoint(Beans.java:483)
		at io.quarkus.arc.processor.BeanInfo.init(BeanInfo.java:581)
		at io.quarkus.arc.processor.BeanDeployment.init(BeanDeployment.java:293)
		... 13 more

```

### Expected behavior

App runs and tests pass

### Actual behavior

App fails to run

### How to Reproduce?

Run https://github.com/quarkusio/quarkus-quickstarts/tree/main/context-propagation-quickstart with latest main

### Output of `uname -a` or `ver`

macOS

### Output of `java -version`

Java 17

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

*   de16f435b6 - (HEAD -> main, upstream/main, origin/main) Merge pull request #33370 from mskacelik/SmallRye-GraphQL-2.2.0 (57 minutes ago) <Jan Martiska>

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/extensions/resteasy-reactive/quarkus-resteasy-reactive-common/deployment/src/main/java/io/quarkus/resteasy/reactive/common/deployment/ResteasyReactiveCommonProcessor.java
+++ b/extensions/resteasy-reactive/quarkus-resteasy-reactive-common/deployment/src/main/java/io/quarkus/resteasy/reactive/common/deployment/ResteasyReactiveCommonProcessor.java
@@ -82,7 +82,7 @@ public class ResteasyReactiveCommonProcessor {
             BuildProducer<AdditionalApplicationArchiveMarkerBuildItem> producer) {
         if (capabilities.isPresent(Capability.RESTEASY) || capabilities.isPresent(Capability.REST_CLIENT)
                 || QuarkusClassLoader.isClassPresentAtRuntime(
-                        "org.jboss.resteasy.reactive.server.injection.JaxrsServerFormUrlEncodedProvider")) { // RESTEasy Classic could be imported via non-Quarkus dependencies
+                        "org.jboss.resteasy.plugins.providers.JaxrsServerFormUrlEncodedProvider")) { // RESTEasy Classic could be imported via non-Quarkus dependencies
             // in this weird case we don't want the providers to be registered automatically as this would lead to multiple bean definitions
             return;
         }

